stream/harfanglab/src/harfanglab.py

```diff
@@ -152,422 +168,526 @@ def _process_message(self, msg):
-            "[DELETE]" if self.harfanglab_indicator_delete else "[DISABLE]"
+    def delete_indicator(self, data, pattern_type, uri, source_list_id):
+        entity = self.helper.api.indicator.read(
+            id=OpenCTIConnectorHelper.get_attribute_in_extension("id", data)
         )
-        self.helper.log_info(
-            f"{indicator_delete} Processing {pattern} indicator"
-            + " {"
-            + OpenCTIConnectorHelper.get_attribute_in_extension("id", data)
-            + "}"
+        config_remove_indicator = (
+            "[DELETE]" if self.harfanglab_remove_indicator else "[DISABLE]"
         )
-
-        indicator_name = self.check_length_indicator_name(data)
-        get_indicator = self._query("get", f"/{uri}/?search={indicator_name}")
-        indicator_matched = self.find_data_name_match(
-            get_indicator["results"], indicator_name
+        self.log_info_process(
+            data, pattern_type, f"{config_remove_indicator}", "indicator"
         )
 
-        if self.harfanglab_indicator_delete is True:
-            if indicator_matched is None:
-                msg_log = f"[DELETE] The searched name of the {pattern} indicator does not exist in HarfangLab"
-                return (self.helper.log_error(msg_log),)
+        if pattern_type == "stix":
+            if entity is not None:
+                observables = entity["observables"]
+                for observable in observables:
+                    observable_matched = self.get_and_match_element(
+                        uri, observable["observable_value"], source_list_id
+                    )
+                    self.process_delete_observable(
+                        data,
+                        entity,
+                        uri,
+                        pattern_type,
+                        config_remove_indicator,
+                        observable_matched,
+                    )
+
+                indicators = self.stix_translation_parser(data, entity)
+                for indicator in indicators:
+                    if indicator["type"] is not None:
+                        indicator_matched = self.get_and_match_element(
+                            uri, indicator["value"], source_list_id
+                        )
+                        self.process_delete_indicator(
+                            data,
+                            entity,
+                            uri,
+                            pattern_type,
+                            config_remove_indicator,
+                            indicator_matched,
+                        )
             else:
-                indicator_id = indicator_matched["id"]
+                observables = self._query(
+                    "get", f"/{uri}/?search={data['id']}&source_id={source_list_id}"
+                )
+                if observables is not None:
+                    for observable in observables["results"]:
+                        self.process_delete_indicator(
```

Comment:
I don't understand why calling `process_delete_indicator` when variables are called observables, can you help me understand ?

Reply:
Indeed this can be confusing, because in the Harfanglab API, it only takes into account the IoC, which only includes types and values, and some metadata so when deleting an indicator in OpenCTI we must delete all the observables and the "OR" linked to this Stix ID indicator enriched at HarfangLab and the process is therefore the same. 

Unfortunately this function also deletes the Yara and Sigma indicators, so we cannot go with a "process_delete_ioc" as you proposed but in fact we would have to find a more consistent name !

Reply:
I see, renaming is not the most important thing to do right now don't worry 😊